src/main/kotlin/no/risc/utils/DifferenceComparer.kt

```diff
@@ -0,0 +1,107 @@
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.jsonArray
+import kotlinx.serialization.json.jsonObject
+
+@Serializable
+data class Difference(
+    val entriesOnLeft: List<String> = listOf(),
+    val entriesOnRight: List<String> = listOf(),
+    val difference: List<String> = listOf(),
+)
+
+/**
+ * Utility class for flattening a RiSc json-object.
+ */
+object FlatMapRiScUtil {
+    /**
+     * Tries to flatten a RiSc map made from json-parsing.
+     *
+     * @param map is a Map made from a json-object
+     * @return A List<String> with each key in a RiSc and its value.
+     */
+    fun flatten(map: Map<String, Any?>): List<String> {
+        fun returnString(
+            key: String,
+            value: String,
+        ): String {
+            return "/$key: $value"
+        }
+
+        return map.flatMap { entry ->
+
+            if (entry == null) {
+                listOf("${entry.key}")
+            }
+
+            when (val value = entry.value) {
+                is List<*> -> {
+                    val list = jsonParseToElement(value.toString()).jsonArray
+
+                    if (list.size == 0) {
+                        listOf(returnString(entry.key, value.toString()))
+                    } else {
+                        val newMap = value.indices.associate { "${entry.key}/$it" to list[it] }
+                        flatten(newMap)
+                    }
+                }
+
+                is JsonElement -> {
+                    val json = jsonParseToElement(value.toString())
+                    when (json) {
+                        is JsonObject -> {
+                            if (json.keys.isEmpty()) {
+                                listOf(returnString(entry.key, value.toString()))
+                            } else {
+                                json.flatMap { flatten(mapOf("${entry.key}/${it.key}" to it.value)) }
+                            }
+                        }
+                        else -> listOf(returnString(entry.key, value.toString()))
+                    }
+                }
+                else -> listOf("${entry.key} - $value is Unknown")
+            }
+        }
+    }
+
+    fun jsonParseToElement(jsonString: String): JsonElement {
+        val json = Json { ignoreUnknownKeys = true }
+        return json.parseToJsonElement(jsonString)
+    }
+}
+
+class DifferenceException(message: String) : Exception(message)
+
+@Throws(DifferenceException::class)
+fun diff(
+    base: String,
+    head: String,
+): Difference {
+    val json = Json { ignoreUnknownKeys = true }
+    try {
+        val baseJsonObject = json.parseToJsonElement(base).jsonObject.toMap()
+        val headJsonObject = json.parseToJsonElement(head).jsonObject.toMap()
+
+        val result1 = FlatMapRiScUtil.flatten(baseJsonObject).associate { it.split(": ").first() to it.split(": ").last() }
```

Comment:
Syntes det er litt vanskelig å skjønne hva som skjer her! Kunne kanskje .associate { it.split(": ").first() to it.split(": ").last() } blitt trukket ut til en extension-function?

`fun List<String>.etForklarendeOrdPåHvaDetteGjør(): TypenPåReturverdi = this.associate { it.split(": ").first() to it.split(": ").last() }`

Reply:
Ja eg er ikkje superfornøgd med all hoppingen mellom Map og List. Føler 50% av utviklingstiden var på å finne riktig `mapOf` eller `listOf` 😅

Kan godt dra ut denne til ein egen funksjon, men vil extention-function bli det beste? Den er jo veldig avhengig av innholdet i listen, så en generisk utvidelse av `List<String>` blir kanskje litt voldsomt?

Reply:
Grunnen til at vi hopper her er fordi vi har ein `List<String>` fra flatmappingen vi utfører, og flatMap jobber kun med Lists. Så ønsker vi å bruke `Maps.difference` funksjonaliteten som krever maps. Og så gir den resultatet sitt tilbake som lister igjen... 🙈